src/Router/DelayedRedirect.purs

```diff
@@ -0,0 +1,37 @@
+module React.Basic.Router.DelayedRedirect where
+
+import Prelude
+
+import Effect (Effect)
+import Effect.Class (liftEffect)
+import Effect.Class.Console as Console
+import Effect.Aff as Aff
+import React.Basic (JSX, element, make)
+import React.Basic as React
+import React.Basic.Router.Foreign as Router
+import React.Basic.Router.Types
+
+type State = { redirectState :: RedirectState }
+
+data RedirectState = Wait | Redirect
+
+delayedRedirect :: forall state. (DelayedRedirectProps state) -> JSX
+delayedRedirect = make component { initialState, render, didMount }
+  where
+    initialState = { redirectState: Wait }
+
+    didMount self@{ props: { delay }, setState, state } = do
+      Aff.launchAff_ do
+        Aff.delay $ Aff.Milliseconds delay
+        liftEffect $ setState _ { redirectState = Redirect }
+
+    render self@{ props: { from, push, to }, state: { redirectState } } =
+      case redirectState of
+        Wait     -> mempty
+        Redirect ->
+          let redirectProps = { from, push, to }
```

Comment:
It bothered me a bit to see `{ from, push, to }` repeated like this, but I couldn't quite see how to express it any better either. So I guess it's just ok.

Reply:
Yes, unfortunately this aspect is different from **Haskell**, AFAIK. But if we find a better way we can always change it later :smiley: